Friendly chat bubbles for livechat widget

itzzritik opened the pull request:
This Pull Request Contains: 
1. New Chat Bubbles UI.
2. Rounded Corners for Avatars.

All can be controlled on a switch in Rocketchat Settings

![image](https://user-images.githubusercontent.com/24320496/123110423-b5ab1c80-d459-11eb-832e-d0f7df2a3e1a.png)

itzzritik committed:
Add friendly chat UI

ear-dev commented:
@ItzzRitik it looks like our tests are failing on this PR. Can you fix please?

Also, @bhardwajaditya can you review please?  Thanks...

bhardwajaditya commented on src/components/Avatar/styles.scss:
```diff
@@ -28,7 +28,7 @@ $avatar-status-indicator-size-large: 14px;
 
 		color: transparent;
-		border-radius: $default-border-radius;
+		border-radius: 50%;
```
Do we want to change this for the current UI as well?

itzzritik replied:
Yes, As @Shailesh351 mentioned this morning, we want to make Avatars round

bhardwajaditya commented on src/components/Messages/Message/index.js:
```diff
@@ -14,6 +14,7 @@ import { MessageContent } from '../MessageContent';
 import { MessageText } from '../MessageText';
 import { MessageTime } from '../MessageTime';
 import { VideoAttachment } from '../VideoAttachment';
+import store from '../../../store/';
```
Please fix lint errors

itzzritik replied:
Sure Thanks

bhardwajaditya reviewed (commented):
PR looks good overall apart from the questions I posted. Couldn't test it on my system since I don't have the setup. @Shailesh351 if you have time you can test this on your system once.

ear-dev commented:
@ItzzRitik @Shailesh351 what does this look like with buttons?  Do we have a design for how we want it to look?  Can you make a suggestion please......... also, please post an image here what he buttons look like at the moment in the chat bubble context.

Shailesh351 commented:
@ItzzRitik @chadgoss This is how it looks with buttons.

<img width="1791" alt="Screenshot 2021-06-25 at 7 49 49 PM" src="https://user-images.githubusercontent.com/18264684/123758108-3e074280-d8dc-11eb-99c7-4b896ea17632.png">

@ItzzRitik Check the rounded corner for the last message. It's looking different because of system messages

ear-dev commented:
@ItzzRitik Can you fix the conflicting file in this PR please?

@Shailesh351 @bhardwajaditya after he fixes the conflicts, this PR is approved?

Thanks.

itzzritik committed:
Fix corners when system message was arround

itzzritik committed:
Merge remote-tracking branch 'origin/develop' into feature/friendly-chat-bubbles

itzzritik committed:
Fix Lint

itzzritik committed:
Add Build Files

itzzritik commented:
@Shailesh351 Your mentioned issue is now fixed.
@ear-dev I have fixed the conflicts in build files.

```I think we're good to go, but would like to have a brief testing before we should proceed with the merge```

### Screenshot of the final change

![image](https://user-images.githubusercontent.com/24320496/126221036-f1e549fd-a97b-4a9e-80ea-cda80ed32b08.png)

bhardwajaditya commented:
> @ItzzRitik Can you fix the conflicting file in this PR please?
> 
> @Shailesh351 @bhardwajaditya after he fixes the conflicts, this PR is approved?
> 
> Thanks.

@ear-dev the PR looks good to me.

Shailesh351 commented on src/components/Messages/MessageList/index.js:
```diff
@@ -147,14 +155,39 @@ export class MessageList extends MemoizedComponent {
+				let sequence = 'mid';
+
+				if (isMyMessage(msg)) {
+					if (!previousMessage || !isMyMessage(previousMessage) || messages[i - 2]?.t) {
```
I didn’t understand why we are using `i-2` here. Isn't it should be `i-1` for the previous message?

itzzritik replied:
for i-1 we get the **time stamp** object instead of the actual message object.
So that's why, Had to do i-2.

Shailesh351 commented on src/components/Messages/MessageList/index.js:
```diff
@@ -147,14 +155,39 @@ export class MessageList extends MemoizedComponent {
 				);
 			}
 
+			const isMyMessage = (msg) => uid && msg?.u && uid === msg?.u._id;
```
Can we move functions related to sequence to some util/helper file? To make it more readable

itzzritik replied:
Yes totally makes sense, Will do it
Thanks!

itzzritik committed:
move Sequence in a helper function

itzzritik committed:
Build files Update

itzzritik commented:
@Shailesh351 @ear-dev 
I have updated the PR
and would like to suggest another round of testing before we merge it.
Thanks!

ear-dev commented:
@ItzzRitik I tested this and I'm seeing a situation where the first chat bubble in the room does not correct its corners upon subsequent messages.  But after the first message, all the others correct themselves.  Can you test that as well and see if you're seeing the same thing?  Thanks. 

![image](https://user-images.githubusercontent.com/28864717/126496714-7abfb7c3-7ceb-470b-ac2d-5f641f9daed9.png)

itzzritik committed:
Improved corner algorithm

itzzritik committed:
Build files update

itzzritik commented:
@ear-dev 
As the Dialogflow bot used to kick in with a welcome message, I could never see this issue.
Thanks for pointing it out.
I've fixed the problem and improved the algorithm for corner detection,
Updated the PR, Please check

ear-dev commented:
@ItzzRitik Now the friendly chat switch is not working at all for me.  Maybe I have a misconfig.......  checking that now......  but can you confirm for me please that this still works for you?  thanks

ear-dev commented:
@ItzzRitik this looks good....  thanks..   Currently we're having an issue with DF that might be related to the new apps-engine in our latest RC version.... so I will wait until that is sorted out before merging.  Thanks!

itzzritik commented:
@ear-dev Thanks for testing!
Sure we should focus on priority task
